_posts/2020-02-10-waku-current-state.md

```diff
@@ -0,0 +1,141 @@
+date:   2020-02-10 12:00:00 +0800
+author: oskarth
+published: false
+permalink: /waku-current-state
+categories: research
+summary: A research log. Waku current state.
+image: /assets/img/TODOFIXME.png
+discuss: https://forum.vac.dev/t/TODOFIXME
+---
+
+TODO: Picture for bottleneck or Waku
+
+# Where we are at with Waku?
+
+Waku is our fork of Whisper where we address the shortcomings of Whisper in an iterative manner. We've seen in previous post that Whisper doesn't scale, and why. Here's an update on where we are at with Waku since that [last post](https://vac.dev/fixing-whisper-with-waku).
+
+## Current state
+
+**Specs:**
+
+We've cut the spec up into three components. These are:
+
+- Waku (main spec), currently in [version 0.2.0](https://specs.vac.dev/waku/waku.html)
+- Waku envelope data field, currently in [version 0.1.0](https://specs.vac.dev/waku/envelope-data-format.html)
+- Waku mailserver, currently in [version 0.2.0](https://specs.vac.dev/waku/mailserver.html)
+
+TODO: Update Waku spec to 0.3
+
+**Clients:**
+
+There are currently two clients that implement Waku, these are [Nimbus](https://github.com/status-im/nimbus/tree/master/waku) in Nim and [status-go](https://github.com/status-im/status-go) in Go.
+
+At the time of this writing the Nimbus client implements the spec fully, but lacks mail server and mail client capability. The status-go client implements everything except bridging mode, which is currently a work in progress.
+
+For more details, see [implementation matrix](https://specs.vac.dev/waku/waku.html#appendix-b-implementation-notes).
+
+In terms of end user applications, work is currently in progress to integrate it into the [Status core app](https://github.com/status-im/status-react/pull/9949) and is expected to be released in their upcoming 1.1 release.
+
+TODO: Fact check this with Adam and Kim
+
+**Simulation:**
+
+We've got a [simulation](https://github.com/status-im/nimbus/tree/master/waku#testing-waku-protocol) in the Nimbus client that verifies - or rather, fails to falsify - the scalability model posted in an [earlier post](https://vac.dev/fixing-whisper-with-waku). More on this below.
+
+## How many users does Waku support?
+
+This is our current understanding of how many users a network running the Waku protocol can support. Specifically in the context of the Status chat app, since that's the most immediate consumer of Waku. It should generalize fairly well to most deployments, but YMMW.
+
+**tldr (for Status app):**
+- beta: 100 DAU
+- v1: 1k DAU
+- v1.1 (waku only): 10k DAU (up to x10 with deployment hotfixes)
+- v1.2 (waku+dns): 100k DAU (can optionally be folded into v1.1)
+
+*Assuming 10 concurrent users = 100 DAU. Estimate uncertainty increases for each order of magnitude until real-world data is observed.*
+
+As far as we know right now, there is one immediate bottleneck, a very likely bottleneck after that, and a third bottleneck that is still conjecture but not unlikely to appear. These are:
```

Comment:
```suggestion
As far as we know right now, these are the bottlenecks we have:
```

Reply:
Hm, there's only one bottleneck at a time though? They don't appear all at once

Reply:
I suppose you edit this below